sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/BloomFilterAggregate.scala

```diff
@@ -55,6 +55,13 @@ case class BloomFilterAggregate(
       Multiply(estimatedNumItemsExpression, Literal(8L)))
   }
 
+  def this(child: Expression, estimatedNumItems: Long) = {
+    this(child, Literal(estimatedNumItems),
+      Literal(math.min(
+        BloomFilter.optimalNumOfBits(estimatedNumItems, estimatedNumItems / 3000000000L.toDouble),
```

Comment:
Can you please explain this `estimatedNumItems / 3000000000L.toDouble`?
Shouldn't we use `BloomFilter.DEFAULT_FPP` which is `0.03`?

Reply:
This is used to reduce the false positive probability if `estimatedNumItems` is small.

Reply:
> estimatedNumItems / 3000000000L.toDouble

Does 3000000000L fit other queries than the TPCDS benchmark?

Reply:
It can benefit all queries if `estimatedNumItems` < 90000000. Especially the number of creationSide rows is small. For example:
```scala
spark.range(1000).selectExpr("id as a", "id as b", "id as c").write.saveAsTable("t1")
spark.range(1000000000).selectExpr("id as x", "id as y", "id as z").write.saveAsTable("t2")

spark.sql("set spark.sql.cbo.enabled=true")
spark.sql("ANALYZE TABLE t1 COMPUTE STATISTICS FOR COLUMNS a, b, c")

spark.sql("select * from t1 left join (select distinct * from t2) t on t1.a = t.y where t1.b > 0").collect()
```
Before this PR | After this PR
-- | --
![image](https://user-images.githubusercontent.com/5399861/187590018-214711c0-80f2-4c4c-9e80-4443a703bb25.png) | ![image](https://user-images.githubusercontent.com/5399861/187589973-941aa3a2-2888-4bd8-9942-2709ed9bf056.png)

Reply:
Thanks @wangyum! Great improvement! I added a comment to the code.